# IGDB 002: IGDB field review

**Per-column EDA on pipeline IGDB artifacts**

Profiles IGDB game columns in `lookups/games.parquet` (Job 1) and Job 2 enriched taxonomy columns (`*_names`, `*_names__use`, `*_names__use_pooled`). Field docs: [`field_docs.py`](../../src/steam_review_ml/igdb/field_docs.py).


# Executive Summary

**Question:**  
For each IGDB game column in our joined parquet, what is the **value shape**, **typical content**, and **likely v2 utility**?

**Result:**  
315 catalog games. Job 1 `lookups/games.parquet` has 17 IGDB fields plus `summary__use` / `storyline__use` (512-d USE, 100% coverage). V2-core FK fields are well populated: genres 100%, themes 98.4%, keywords 91.1%, game_modes 100%, player_perspectives 95.9%. Franchises are sparse (30%) — not a V2a candidate. Job 2 enriched adds 15 taxonomy columns (`*_names`, `*_names__use`, `*_names__use_pooled`); pooled vectors are 512-d L2-normalized and mirror FK coverage per field. Manual mock for Wallpaper Engine (`431960`) resolves (e.g. genres → Indie, Simulator).

**Recommendation / Decision:**  
**Proceed with V2a.** Wire metadata overlap on FK id sets (`genres`, `themes`, `keywords`, `game_modes`, `player_perspectives`) for category-bridge retrieval. Optionally ablate `*_names__use_pooled` cosine vs Jaccard later. Use `summary__use` for V2b text similarity. Defer `franchises` and non-core id-list fields.


# Data Sources

**Job 1:** `artifacts/igdb/lookups/games.parquet` (`recs_job_igdb_games.py`)

**Job 2:** `artifacts/igdb/igdb_games__enriched.parquet` (`recs_job_igdb_games_enriched.py`) — taxonomy `*_names`, `*_names__use`, `*_names__use_pooled`

**Join context:** `artifacts/igdb/igdb_join_report.json` (optional)

**Pipeline refresh:**
```bash
python scripts/recs_job_igdb_games.py configs/recs_job_igdb_games.json
python scripts/recs_job_igdb_games_enriched.py configs/recs_job_igdb_games_enriched.json
```


# Notebook Roadmap

1. Setup + load Job 1 / Job 2 parquets
2. Overview table (Job 1 IGDB fields)
3. Per-field profiles (Job 1)
4. Job 2 enriched taxonomy columns (`*_names__use_pooled`, etc.)
5. Key Findings (manual)


# Analysis


## Setup


In [1]:
from __future__ import annotations

import json
from pathlib import Path
from typing import Any

import numpy as np
import pandas as pd
from IPython.display import Markdown, display

from steam_review_ml.igdb.constants import (
    IGDB_GAMES_ENRICHED_PARQUET,
    IGDB_GAMES_PARQUET,
    STEAM_JOIN_COLS,
    TAXONOMY_RESOLVE_FIELDS,
    V2_CORE_FIELDS,
)
from steam_review_ml.igdb.entity_lookup import (
    taxonomy_names_column,
    taxonomy_names_embedding_column,
    taxonomy_names_embedding_pooled_column,
)
from steam_review_ml.igdb.field_docs import GAME_FIELD_DOCS, V2_FIELD_NOTES

REPO_ROOT = next(p for p in [Path.cwd(), *Path.cwd().parents] if (p / "pyproject.toml").is_file())
IGDB_DIR = REPO_ROOT / "artifacts/igdb"
PARQUET_PATH = IGDB_DIR / IGDB_GAMES_PARQUET
ENRICHED_PATH = IGDB_DIR / IGDB_GAMES_ENRICHED_PARQUET
REPORT_PATH = IGDB_DIR / "igdb_join_report.json"

SAMPLE_APP_IDS = [753420, 646910, 512900, 431960]

print(f"REPO_ROOT={REPO_ROOT}")


REPO_ROOT=/home/ryanr/workspace/steam_recommendations


In [2]:
if not PARQUET_PATH.is_file():
    raise FileNotFoundError(
        f"Missing {PARQUET_PATH}. Run:\n"
        "  python scripts/recs_job_igdb_games.py configs/recs_job_igdb_games.json"
    )

df = pd.read_parquet(PARQUET_PATH)
join_report = json.loads(REPORT_PATH.read_text(encoding="utf-8")) if REPORT_PATH.is_file() else {}

enriched_df: pd.DataFrame | None = None
ENRICHED_TAXONOMY_FIELDS: list[str] = []
if ENRICHED_PATH.is_file():
    enriched_df = pd.read_parquet(ENRICHED_PATH)
    ENRICHED_TAXONOMY_FIELDS = [
        col
        for field in TAXONOMY_RESOLVE_FIELDS
        for col in (
            taxonomy_names_column(field),
            taxonomy_names_embedding_column(field),
            taxonomy_names_embedding_pooled_column(field),
        )
        if col in enriched_df.columns
    ]
else:
    print(
        f"Skip enriched profiles: missing {ENRICHED_PATH}\n"
        "  python scripts/recs_job_igdb_games_enriched.py configs/recs_job_igdb_games_enriched.json"
    )

IGDB_GAME_FIELDS = sorted(c for c in df.columns if c not in STEAM_JOIN_COLS and c != "igdb_name")
absent_from_parquet = sorted(set(GAME_FIELD_DOCS) - set(IGDB_GAME_FIELDS) - {"name"})

print(f"rows={len(df)} cols={len(df.columns)}")
print(f"IGDB_GAME_FIELDS ({len(IGDB_GAME_FIELDS)}): {IGDB_GAME_FIELDS}")
if enriched_df is not None:
    print(f"enriched rows={len(enriched_df)} enriched_taxonomy_fields={ENRICHED_TAXONOMY_FIELDS}")
print(f"catalog match rate: {join_report.get('match_rate', 'n/a')}")
if absent_from_parquet:
    print(f"Fields in GAME_FIELD_DOCS but not in parquet: {absent_from_parquet}")


rows=315 cols=22
IGDB_GAME_FIELDS (17): ['age_ratings', 'collections', 'franchises', 'game_engines', 'game_modes', 'game_type', 'genres', 'involved_companies', 'keywords', 'multiplayer_modes', 'player_perspectives', 'storyline', 'storyline__use', 'summary', 'summary__use', 'tags', 'themes']
enriched rows=315 enriched_taxonomy_fields=['genres_names', 'genres_names__use', 'genres_names__use_pooled', 'themes_names', 'themes_names__use', 'themes_names__use_pooled', 'keywords_names', 'keywords_names__use', 'keywords_names__use_pooled', 'game_modes_names', 'game_modes_names__use', 'game_modes_names__use_pooled', 'player_perspectives_names', 'player_perspectives_names__use', 'player_perspectives_names__use_pooled']
catalog match rate: 1.0
Fields in GAME_FIELD_DOCS but not in parquet: ['aggregated_rating', 'aggregated_rating_count', 'alternative_names', 'artworks', 'bundles', 'category', 'checksum', 'collection', 'cover', 'created_at', 'dlcs', 'expanded_games', 'expansions', 'external_games', 

In [3]:
def igdb_field_doc(field: str) -> dict[str, str]:
    doc = GAME_FIELD_DOCS.get(field, {})
    return {
        "igdb_type": doc.get("type", ""),
        "igdb_description": doc.get("description", ""),
        "deprecated": doc.get("deprecated", False),
        "v2_note": V2_FIELD_NOTES.get(field, ""),
    }


def field_populated(value: Any) -> bool:
    if value is None:
        return False
    try:
        if pd.isna(value):
            return False
    except (TypeError, ValueError):
        pass
    if isinstance(value, str):
        return bool(value.strip())
    if isinstance(value, (list, dict, set, tuple, np.ndarray)):
        return len(value) > 0
    return True


def _as_list(value: Any) -> list[Any]:
    if value is None:
        return []
    if isinstance(value, np.ndarray):
        return value.tolist()
    if isinstance(value, (list, tuple, set)):
        return list(value)
    try:
        if pd.isna(value):
            return []
    except (TypeError, ValueError):
        pass
    return [value]


def _is_embedding(value: Any) -> bool:
    if not isinstance(value, np.ndarray):
        return False
    return value.ndim == 1 and np.issubdtype(value.dtype, np.floating)


def value_kind(value: Any) -> str:
    if not field_populated(value):
        return "empty"
    if isinstance(value, str):
        return "string"
    if isinstance(value, (int, np.integer)):
        return "integer"
    if isinstance(value, (float, np.floating)):
        return "float"
    if _is_embedding(value):
        return "embedding"
    if isinstance(value, np.ndarray):
        if value.ndim == 0:
            return value_kind(value.item())
        return "id_list" if np.issubdtype(value.dtype, np.integer) else "array"
    if isinstance(value, list):
        if value and _is_embedding(value[0]):
            return "embedding_list"
        return "id_list" if value and all(isinstance(x, (int, np.integer)) for x in value) else "list"
    return type(value).__name__


def profile_column(series: pd.Series) -> dict[str, Any]:
    populated = series.map(field_populated)
    populated_s = series.loc[populated]
    kinds = populated_s.map(value_kind)
    kind_mode = kinds.mode().iloc[0] if len(kinds) else "empty"

    doc = igdb_field_doc(str(series.name))
    out: dict[str, Any] = {
        "field": series.name,
        "dtype": str(series.dtype),
        "coverage_pct": round(float(populated.mean()) * 100, 2),
        "value_kind": kind_mode,
        "v2_core": series.name in V2_CORE_FIELDS or str(series.name).endswith("__use_pooled"),
        "igdb_type": doc["igdb_type"],
        "igdb_description": doc["igdb_description"],
        "v2_note": doc["v2_note"],
    }

    if kind_mode == "string":
        lens = populated_s.astype(str).str.len()
        out["len_median"] = int(lens.median())
        out["len_p95"] = int(lens.quantile(0.95))
    elif kind_mode in {"integer", "float"}:
        nums = pd.to_numeric(populated_s, errors="coerce").dropna()
        if len(nums):
            out["min"] = float(nums.min())
            out["median"] = float(nums.median())
            out["max"] = float(nums.max())
    elif kind_mode == "id_list":
        lengths = populated_s.map(lambda v: len(_as_list(v)))
        out["list_len_median"] = float(lengths.median())
        out["list_len_p95"] = float(lengths.quantile(0.95))
        out["unique_ids"] = int(pd.Series([x for v in populated_s for x in _as_list(v)]).nunique())
    elif kind_mode == "list":
        lengths = populated_s.map(lambda v: len(_as_list(v)))
        out["list_len_median"] = float(lengths.median())
        out["list_len_p95"] = float(lengths.quantile(0.95))
    elif kind_mode == "embedding":
        dims = populated_s.map(lambda v: int(np.asarray(v).shape[0]))
        out["embed_dim"] = int(dims.mode().iloc[0]) if len(dims) else None
    elif kind_mode == "embedding_list":
        lengths = populated_s.map(lambda v: len(v))
        out["list_len_median"] = float(lengths.median())
        out["list_len_p95"] = float(lengths.quantile(0.95))
        dims = populated_s.map(lambda v: int(np.asarray(v[0]).shape[0]) if len(v) else 0)
        out["embed_dim"] = int(dims.mode().iloc[0]) if len(dims) else None

    return out


def format_sample(value: Any, *, max_len: int = 240) -> str:
    if not field_populated(value):
        return "<empty>"
    if isinstance(value, str):
        text = value.replace("\n", " ")
        return text if len(text) <= max_len else text[: max_len - 3] + "..."
    if _is_embedding(value):
        arr = np.asarray(value, dtype=np.float32)
        norm = float(np.linalg.norm(arr))
        return f"ndarray(shape={arr.shape[0]}, norm={norm:.3f})"
    if isinstance(value, list) and value and _is_embedding(value[0]):
        dim = int(np.asarray(value[0]).shape[0])
        return f"list[{len(value)}]×ndarray({dim},)"
    if isinstance(value, np.ndarray):
        return f"array({value.tolist()})"
    return repr(value)[:max_len]


def display_field_profile(data: pd.DataFrame, field: str) -> None:
    series = data[field]
    prof = profile_column(series)
    doc = igdb_field_doc(field)
    v2_tag = " **V2_CORE**" if prof.get("v2_core") else ""
    dep_tag = " *(deprecated)*" if doc.get("deprecated") else ""
    display(Markdown(f"### `{field}`{v2_tag}{dep_tag}"))
    if doc.get("igdb_type"):
        display(Markdown(f"**IGDB type:** `{doc['igdb_type']}`"))
    if doc.get("igdb_description"):
        display(Markdown(f"**IGDB description:** {doc['igdb_description']}"))
    if str(field).endswith("__use_pooled"):
        display(Markdown("**Job 2 note:** Mean-pooled, L2-normalized USE vector over resolved taxonomy entity names."))
    elif doc.get("v2_note"):
        display(Markdown(f"**V2 note:** {doc['v2_note']}"))

    summary_rows = {
        k: v
        for k, v in prof.items()
        if k not in {"field", "igdb_type", "igdb_description", "v2_note", "v2_core"}
    }
    display(pd.DataFrame([summary_rows]))

    sample_df = (
        data.loc[data["app_id"].isin(SAMPLE_APP_IDS), ["app_id", "app_name", field]]
        .sort_values("app_id")
        .assign(sample=lambda d: d[field].map(format_sample))
        .drop(columns=[field])
    )
    display(sample_df)

    extra = data.loc[series.map(field_populated), field].head(3)
    print("Additional samples:")
    for app_id, val in zip(data.loc[extra.index, "app_id"], extra):
        print(f"  app_id={app_id}: {format_sample(val)}")


## Overview


In [4]:
overview = pd.DataFrame([profile_column(df[c]) for c in IGDB_GAME_FIELDS])
overview = overview.sort_values(["v2_core", "coverage_pct"], ascending=[False, False]).reset_index(drop=True)
display(overview)

display(overview.loc[overview["v2_core"], ["field", "coverage_pct", "value_kind", "igdb_type", "v2_note"]])


,field,dtype,coverage_pct,value_kind,v2_core,igdb_type,igdb_description,v2_note,list_len_median,list_len_p95,unique_ids,min,median,max,len_median,len_p95,embed_dim
0,game_modes,object,100.00,id_list,True,Array of Game Mode IDs,Modes of gameplay,V2a — Jaccard on game mode ID sets.,2.0,4.0,6.0,NaN,NaN,NaN,NaN,NaN,NaN
1,genres,object,100.00,id_list,True,Array of Genre IDs,Genres of the game,V2a — Jaccard on genre ID sets (entity lookup ...,3.0,5.0,20.0,NaN,NaN,NaN,NaN,NaN,NaN
2,summary,str,100.00,string,True,String,A description of the game,"V2b — USE dot(query_review, summary). Ready wi...",NaN,NaN,NaN,NaN,NaN,NaN,357.0,903.0,NaN
3,themes,object,98.41,id_list,True,Array of Theme IDs,Themes of the game,V2a — Jaccard on theme ID sets.,2.0,5.0,22.0,NaN,NaN,NaN,NaN,NaN,NaN
4,player_perspectives,object,95.87,id_list,True,Array of Player Perspective IDs,The main perspective of the player,V2a — Jaccard on perspective ID sets.,1.0,2.0,7.0,NaN,NaN,NaN,NaN,NaN,NaN
5,keywords,object,91.11,id_list,True,Array of Keyword IDs,Associated keywords,V2a — Jaccard on keyword ID sets; ~91% coverag...,12.0,84.0,1707.0,NaN,NaN,NaN,NaN,NaN,NaN
6,franchises,object,29.84,id_list,True,Array of Franchise IDs,Other franchises the game belongs to,V2a — Jaccard on franchise ID sets; sparse (~3...,1.0,2.0,91.0,NaN,NaN,NaN,NaN,NaN,NaN
7,game_type,int64,100.00,integer,False,Reference ID for Game Type,The type of game,Filter DLC/expansion/bundle noise before ranke...,NaN,NaN,NaN,0.0,0.0,11.0,NaN,NaN,NaN
8,storyline__use,object,100.00,embedding,False,,,,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,512.0
9,summary__use,object,100.00,embedding,False,,,,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,512.0


,field,coverage_pct,value_kind,igdb_type,v2_note
0,game_modes,100.00,id_list,Array of Game Mode IDs,V2a — Jaccard on game mode ID sets.
1,genres,100.00,id_list,Array of Genre IDs,V2a — Jaccard on genre ID sets (entity lookup ...
2,summary,100.00,string,String,"V2b — USE dot(query_review, summary). Ready wi..."
3,themes,98.41,id_list,Array of Theme IDs,V2a — Jaccard on theme ID sets.
4,player_perspectives,95.87,id_list,Array of Player Perspective IDs,V2a — Jaccard on perspective ID sets.
5,keywords,91.11,id_list,Array of Keyword IDs,V2a — Jaccard on keyword ID sets; ~91% coverag...
6,franchises,29.84,id_list,Array of Franchise IDs,V2a — Jaccard on franchise ID sets; sparse (~3...


## Per-field review

One subsection per IGDB game column.


In [5]:
for field in IGDB_GAME_FIELDS:
    display_field_profile(df, field)
    print()


### `age_ratings`

**IGDB type:** `Array of Age Rating IDs`

**IGDB description:** The PEGI rating

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,85.08,id_list,5.0,7.0,1328


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,"array([125538, 195813, 39086, 181520, 57827, 1..."
1,646910,The Crew 2,"array([111035, 111034, 91834, 92244, 215725, 2..."
0,753420,Dungreed,"array([119531, 55417, 98238, 187890])"


Additional samples:
  app_id=753420: array([119531, 55417, 98238, 187890])
  app_id=646910: array([111035, 111034, 91834, 92244, 215725, 215726, 25509])
  app_id=512900: array([125538, 195813, 39086, 181520, 57827, 120997])



### `collections`

**IGDB type:** `Array of Collection IDs`

**IGDB description:** The collections that this game is in

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,65.08,id_list,1.0,1.8,183


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,array([11919])
1,646910,The Crew 2,array([2719])
0,753420,Dungreed,<empty>


Additional samples:
  app_id=646910: array([2719])
  app_id=512900: array([11919])
  app_id=637090: array([2091])



### `franchises` **V2_CORE**

**IGDB type:** `Array of Franchise IDs`

**IGDB description:** Other franchises the game belongs to

**V2 note:** V2a — Jaccard on franchise ID sets; sparse (~30% coverage) on our catalog.

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,29.84,id_list,1.0,2.0,91


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,<empty>
1,646910,The Crew 2,<empty>
0,753420,Dungreed,<empty>


Additional samples:
  app_id=613830: array([1844])
  app_id=748490: array([855])
  app_id=825630: array([842])



### `game_engines`

**IGDB type:** `Array of Game Engine IDs`

**IGDB description:** The game engine used in this game

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,78.1,id_list,1.0,1.0,102


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,array([13])
1,646910,The Crew 2,array([118])
0,753420,Dungreed,array([13])


Additional samples:
  app_id=753420: array([13])
  app_id=646910: array([118])
  app_id=512900: array([13])



### `game_modes` **V2_CORE**

**IGDB type:** `Array of Game Mode IDs`

**IGDB description:** Modes of gameplay

**V2 note:** V2a — Jaccard on game mode ID sets.

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,100.0,id_list,2.0,4.0,6


,app_id,app_name,sample
314,431960,Wallpaper Engine,array([1])
2,512900,Streets of Rogue,"array([1, 2, 3, 4])"
1,646910,The Crew 2,"array([1, 2, 3, 5])"
0,753420,Dungreed,array([1])


Additional samples:
  app_id=753420: array([1])
  app_id=646910: array([1, 2, 3, 5])
  app_id=512900: array([1, 2, 3, 4])



### `game_type`

**IGDB type:** `Reference ID for Game Type`

**IGDB description:** The type of game

**V2 note:** Filter DLC/expansion/bundle noise before ranker features.

,dtype,coverage_pct,value_kind,min,median,max
0,int64,100.0,integer,0.0,0.0,11.0


,app_id,app_name,sample
314,431960,Wallpaper Engine,0
2,512900,Streets of Rogue,0
1,646910,The Crew 2,0
0,753420,Dungreed,0


Additional samples:
  app_id=753420: 0
  app_id=646910: 0
  app_id=512900: 0



### `genres` **V2_CORE**

**IGDB type:** `Array of Genre IDs`

**IGDB description:** Genres of the game

**V2 note:** V2a — Jaccard on genre ID sets (entity lookup optional for human QA).

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,100.0,id_list,3.0,5.0,20


,app_id,app_name,sample
314,431960,Wallpaper Engine,"array([32, 13])"
2,512900,Streets of Rogue,"array([5, 12, 25, 31, 32])"
1,646910,The Crew 2,array([10])
0,753420,Dungreed,"array([8, 31, 32])"


Additional samples:
  app_id=753420: array([8, 31, 32])
  app_id=646910: array([10])
  app_id=512900: array([5, 12, 25, 31, 32])



### `involved_companies`

**IGDB type:** `Array of Involved Company IDs`

**IGDB description:** Companies who developed this game

**V2 note:** Developer/publisher IDs; optional metadata signal.

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,99.05,id_list,2.0,5.0,791


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,"array([116692, 116693, 277284])"
1,646910,The Crew 2,"array([50782, 141206, 202299])"
0,753420,Dungreed,"array([72008, 128578])"


Additional samples:
  app_id=753420: array([72008, 128578])
  app_id=646910: array([50782, 141206, 202299])
  app_id=512900: array([116692, 116693, 277284])



### `keywords` **V2_CORE**

**IGDB type:** `Array of Keyword IDs`

**IGDB description:** Associated keywords

**V2 note:** V2a — Jaccard on keyword ID sets; ~91% coverage on our catalog.

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,91.11,id_list,12.0,84.0,1707


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,"array([416, 577, 1033, 1980, 4154, 4466, 4882,..."
1,646910,The Crew 2,"array([155, 613, 778, 2071, 4357])"
0,753420,Dungreed,<empty>


Additional samples:
  app_id=646910: array([155, 613, 778, 2071, 4357])
  app_id=512900: array([416, 577, 1033, 1980, 4154, 4466, 4882, 17292, 26969])
  app_id=637090: array([69, 167, 415, 575, 1107, 1317, 1379, 1448, 2425, 4134, 4248, 4250, 4272, 4882, 4886, 5323, 5379, 5453, 6304, 6699, 9357, 49331])



### `multiplayer_modes`

**IGDB type:** `Array of Multiplayer Mode IDs`

**IGDB description:** Multiplayer modes for this game

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,41.27,id_list,1.0,3.0,196


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,array([3937])
1,646910,The Crew 2,array([8291])
0,753420,Dungreed,<empty>


Additional samples:
  app_id=646910: array([8291])
  app_id=512900: array([3937])
  app_id=214950: array([7273])



### `player_perspectives` **V2_CORE**

**IGDB type:** `Array of Player Perspective IDs`

**IGDB description:** The main perspective of the player

**V2 note:** V2a — Jaccard on perspective ID sets.

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,95.87,id_list,1.0,2.0,7


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,array([3])
1,646910,The Crew 2,array([2])
0,753420,Dungreed,array([4])


Additional samples:
  app_id=753420: array([4])
  app_id=646910: array([2])
  app_id=512900: array([3])



### `storyline`

**IGDB type:** `String`

**IGDB description:** A short description of a game's story

**V2 note:** Extra text field (not summary); ~50% coverage; optional V2b supplement.

,dtype,coverage_pct,value_kind,len_median,len_p95
0,str,49.84,string,566,2597


,app_id,app_name,sample
314,431960,Wallpaper Engine,Use stunning live wallpapers on your desktop. ...
2,512900,Streets of Rogue,<empty>
1,646910,The Crew 2,"The game features a nonlinear story, that foll..."
0,753420,Dungreed,<empty>


Additional samples:
  app_id=646910: The game features a nonlinear story, that follows the unnamed player character as they become a racing icon in the United States by winning in all racing disciplines available in the game. There are four disciplines: Street Racing, Off R...
  app_id=613830: A chance encounter amid the festivities of Guardia's Millennial Fair in Leene Square introduces our young hero, Crono, to a girl by the name of Marle. Deciding to explore the fair together, the two soon find themselves at an exhibition o...
  app_id=748490: The Legend of Heroes: Trails of Cold Steel II picks up one month after the decisive collision that changed the fate of the entire nation of Erebonia. The speedy, tactical turn-based combat with the newly-developed “ARCUS” system returns,...



### `storyline__use`

,dtype,coverage_pct,value_kind,embed_dim
0,object,100.0,embedding,512


,app_id,app_name,sample
314,431960,Wallpaper Engine,"ndarray(shape=512, norm=1.000)"
2,512900,Streets of Rogue,"ndarray(shape=512, norm=1.000)"
1,646910,The Crew 2,"ndarray(shape=512, norm=1.000)"
0,753420,Dungreed,"ndarray(shape=512, norm=1.000)"


Additional samples:
  app_id=753420: ndarray(shape=512, norm=1.000)
  app_id=646910: ndarray(shape=512, norm=1.000)
  app_id=512900: ndarray(shape=512, norm=1.000)



### `summary` **V2_CORE**

**IGDB type:** `String`

**IGDB description:** A description of the game

**V2 note:** V2b — USE dot(query_review, summary). Ready without entity lookup.

,dtype,coverage_pct,value_kind,len_median,len_p95
0,str,100.0,string,357,903


,app_id,app_name,sample
314,431960,Wallpaper Engine,Wallpaper Engine enables you to use live wallp...
2,512900,Streets of Rogue,Streets of Rogue is a top-down rogue-lite with...
1,646910,The Crew 2,The newest iteration in the revolutionary fran...
0,753420,Dungreed,Dungreed is 2D side-scrolling action game with...


Additional samples:
  app_id=753420: Dungreed is 2D side-scrolling action game with a Rogue-LITE element. You'll explore the ever-changing dungeon that destroyed everything in the village. Kill enemies, Use various weapons, spells, and eat food to defeat evil in the dungeon!
  app_id=646910: The newest iteration in the revolutionary franchise, The Crew 2 captures the thrill of the American motorsports spirit in one of the most exhilarating open worlds ever created. Welcome to Motornation, a huge, varied, action-packed, and b...
  app_id=512900: Streets of Rogue is a top-down rogue-lite with an emphasis on player agency and freedom. It combines shooting, stealth, and role-playing elements in a procedurally generated city.  Rather than taking place in a dungeon, the game is set i...



### `summary__use`

,dtype,coverage_pct,value_kind,embed_dim
0,object,100.0,embedding,512


,app_id,app_name,sample
314,431960,Wallpaper Engine,"ndarray(shape=512, norm=1.000)"
2,512900,Streets of Rogue,"ndarray(shape=512, norm=1.000)"
1,646910,The Crew 2,"ndarray(shape=512, norm=1.000)"
0,753420,Dungreed,"ndarray(shape=512, norm=1.000)"


Additional samples:
  app_id=753420: ndarray(shape=512, norm=1.000)
  app_id=646910: ndarray(shape=512, norm=1.000)
  app_id=512900: ndarray(shape=512, norm=1.000)



### `tags`

**IGDB type:** `Array of Tag Numbers`

**IGDB description:** Related entities in the IGDB API

**V2 note:** V2a candidate — related-entity tag numbers; noisier than genres/themes.

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,99.68,id_list,16.0,91.0,1744


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,"array([1, 23, 27, 33, 268435461, 268435468, 26..."
1,646910,The Crew 2,"array([1, 38, 268435466, 536871067, 536871525,..."
0,753420,Dungreed,"array([1, 268435464, 268435487, 268435488])"


Additional samples:
  app_id=753420: array([1, 268435464, 268435487, 268435488])
  app_id=646910: array([1, 38, 268435466, 536871067, 536871525, 536871690, 536872983, 536875269])
  app_id=512900: array([1, 23, 27, 33, 268435461, 268435468, 268435481, 268435487, 268435488, 536871328, 536871489, 536871945, 536872892, 536875066, 536875378, 536875794, 536888204, 536897881])



### `themes` **V2_CORE**

**IGDB type:** `Array of Theme IDs`

**IGDB description:** Themes of the game

**V2 note:** V2a — Jaccard on theme ID sets.

,dtype,coverage_pct,value_kind,list_len_median,list_len_p95,unique_ids
0,object,98.41,id_list,2.0,5.0,22


,app_id,app_name,sample
314,431960,Wallpaper Engine,array([38])
2,512900,Streets of Rogue,"array([1, 23, 27, 33])"
1,646910,The Crew 2,"array([1, 38])"
0,753420,Dungreed,array([1])


Additional samples:
  app_id=753420: array([1])
  app_id=646910: array([1, 38])
  app_id=512900: array([1, 23, 27, 33])



## Job 2 enriched taxonomy columns

Resolved from `igdb_games__enriched.parquet`: `{field}_names`, `{field}_names__use` (per-entity vectors), `{field}_names__use_pooled` (mean-pooled field vector).

In [6]:
if enriched_df is None or not ENRICHED_TAXONOMY_FIELDS:
    print("No enriched taxonomy columns to profile.")
else:
    enriched_overview = pd.DataFrame(
        [profile_column(enriched_df[c]) for c in ENRICHED_TAXONOMY_FIELDS]
    )
    enriched_overview = enriched_overview.sort_values(
        ["field"], ascending=True
    ).reset_index(drop=True)
    display(enriched_overview)

    pooled_cols = [c for c in ENRICHED_TAXONOMY_FIELDS if c.endswith("__use_pooled")]
    if pooled_cols:
        display(
            enriched_overview.loc[
                enriched_overview["field"].isin(pooled_cols),
                ["field", "coverage_pct", "value_kind", "embed_dim"],
            ]
        )

    for field in ENRICHED_TAXONOMY_FIELDS:
        display_field_profile(enriched_df, field)
        print()

,field,dtype,coverage_pct,value_kind,v2_core,igdb_type,igdb_description,v2_note,embed_dim
0,game_modes_names,object,100.00,array,False,,,,NaN
1,game_modes_names__use,object,100.00,array,False,,,,NaN
2,game_modes_names__use_pooled,object,100.00,embedding,True,,,,512.0
3,genres_names,object,100.00,array,False,,,,NaN
4,genres_names__use,object,100.00,array,False,,,,NaN
5,genres_names__use_pooled,object,100.00,embedding,True,,,,512.0
6,keywords_names,object,91.11,array,False,,,,NaN
7,keywords_names__use,object,91.11,array,False,,,,NaN
8,keywords_names__use_pooled,object,91.11,embedding,True,,,,512.0
9,player_perspectives_names,object,95.87,array,False,,,,NaN


,field,coverage_pct,value_kind,embed_dim
2,game_modes_names__use_pooled,100.00,embedding,512.0
5,genres_names__use_pooled,100.00,embedding,512.0
8,keywords_names__use_pooled,91.11,embedding,512.0
11,player_perspectives_names__use_pooled,95.87,embedding,512.0
14,themes_names__use_pooled,98.41,embedding,512.0


### `genres_names`

,dtype,coverage_pct,value_kind
0,object,100.0,array


,app_id,app_name,sample
314,431960,Wallpaper Engine,"array(['Indie', 'Simulator'])"
2,512900,Streets of Rogue,"array(['Shooter', 'Role-playing (RPG)', ""Hack ..."
1,646910,The Crew 2,array(['Racing'])
0,753420,Dungreed,"array(['Platform', 'Adventure', 'Indie'])"


Additional samples:
  app_id=753420: array(['Platform', 'Adventure', 'Indie'])
  app_id=646910: array(['Racing'])
  app_id=512900: array(['Shooter', 'Role-playing (RPG)', "Hack and slash/Beat 'em up", 'Adventure', 'Indie'])



### `genres_names__use`

,dtype,coverage_pct,value_kind
0,object,100.0,array


,app_id,app_name,sample
314,431960,Wallpaper Engine,"array([array([-0.07399165, 0.04525759, 0.076..."
2,512900,Streets of Rogue,"array([array([-3.33105661e-02, 3.96502987e-02..."
1,646910,The Crew 2,"array([array([-1.04813194e-02, -1.08511504e-02..."
0,753420,Dungreed,"array([array([-3.24164629e-02, -1.16465865e-02..."


Additional samples:
  app_id=753420: array([array([-3.24164629e-02, -1.16465865e-02, -3.52083705e-02,  6.60861582e-02,
       -6.82177842e-02, -5.01222946e-02,  4.04949076e-02,  4.51646894e-02,
        5.99597730e-02, -5.23042567e-02, -6.68108184e-03,  6.37105778e-02,
       -5.99944107e-02, -4.83499840e-02, -3.43338251e-02,  5.52915633e-02,
        1.00388797e-02,  2.50155851e-02, -2.82744560e-02,  2.91051343e-02,
        6.23185858e-02,  5.51786162e-02, -1.51140522e-02, -1.92491189e-02,
        9.77447256e-03,  2.71013137e-02, -1.87551063e-02, -6.96919337e-02,
       -2.70267352e-02, -3.68802585e-02, -6.30523488e-02, -3.12220044e-02,
       -7.09714973e-03,  7.78370351e-02,  2.12080162e-02, -3.71033922e-02,
        4.96508591e-02,  7.58511573e-02,  3.19386274e-02,  7.76277781e-02,
        5.52592278e-02,  8.11665878e-02, -5.63502274e-02, -7.81770200e-02,
        2.69493312e-02,  6.84107691e-02,  1.68583216e-03, -2.29211878e-02,
       -7.79724568e-02,  8.04600194e-02,  1.68284681e-02

### `genres_names__use_pooled` **V2_CORE**

**Job 2 note:** Mean-pooled, L2-normalized USE vector over resolved taxonomy entity names.

,dtype,coverage_pct,value_kind,embed_dim
0,object,100.0,embedding,512


,app_id,app_name,sample
314,431960,Wallpaper Engine,"ndarray(shape=512, norm=1.000)"
2,512900,Streets of Rogue,"ndarray(shape=512, norm=1.000)"
1,646910,The Crew 2,"ndarray(shape=512, norm=1.000)"
0,753420,Dungreed,"ndarray(shape=512, norm=1.000)"


Additional samples:
  app_id=753420: ndarray(shape=512, norm=1.000)
  app_id=646910: ndarray(shape=512, norm=1.000)
  app_id=512900: ndarray(shape=512, norm=1.000)



### `themes_names`

,dtype,coverage_pct,value_kind
0,object,98.41,array


,app_id,app_name,sample
314,431960,Wallpaper Engine,array(['Open world'])
2,512900,Streets of Rogue,"array(['Action', 'Stealth', 'Comedy', 'Sandbox'])"
1,646910,The Crew 2,"array(['Action', 'Open world'])"
0,753420,Dungreed,array(['Action'])


Additional samples:
  app_id=753420: array(['Action'])
  app_id=646910: array(['Action', 'Open world'])
  app_id=512900: array(['Action', 'Stealth', 'Comedy', 'Sandbox'])



### `themes_names__use`

,dtype,coverage_pct,value_kind
0,object,98.41,array


,app_id,app_name,sample
314,431960,Wallpaper Engine,"array([array([ 0.00401841, -0.02488704, -0.018..."
2,512900,Streets of Rogue,"array([array([-0.05623418, -0.00843579, -0.050..."
1,646910,The Crew 2,"array([array([-0.05623418, -0.00843579, -0.050..."
0,753420,Dungreed,"array([array([-0.05623418, -0.00843579, -0.050..."


Additional samples:
  app_id=753420: array([array([-0.05623418, -0.00843579, -0.0502703 ,  0.00560191, -0.04610413,
        0.07423158, -0.06005892,  0.01882012,  0.01781887,  0.00748416,
        0.02844358, -0.00650683,  0.01309401,  0.02006292, -0.00651936,
       -0.00169   ,  0.03554761,  0.01583059,  0.02827195,  0.06353868,
        0.00316711,  0.01930034,  0.04640906, -0.03143368, -0.00550876,
        0.04017363, -0.02396203,  0.013983  , -0.07261098,  0.0203326 ,
       -0.03197872, -0.00825811,  0.01260537, -0.00177444, -0.0065135 ,
        0.01046905,  0.04239368, -0.00821904,  0.00911604, -0.03528917,
       -0.04264298,  0.05964223, -0.04318275, -0.06429168, -0.0440727 ,
        0.06417164,  0.027254  , -0.06433674,  0.03436488, -0.01154627,
        0.05735147, -0.0247459 ,  0.0411249 ,  0.07606389, -0.0645652 ,
        0.04168431, -0.02519169, -0.00936455,  0.08496702,  0.0039042 ,
        0.04405007,  0.06669956,  0.02150888,  0.00935683, -0.04458494,
        0.0103469 , 

### `themes_names__use_pooled` **V2_CORE**

**Job 2 note:** Mean-pooled, L2-normalized USE vector over resolved taxonomy entity names.

,dtype,coverage_pct,value_kind,embed_dim
0,object,98.41,embedding,512


,app_id,app_name,sample
314,431960,Wallpaper Engine,"ndarray(shape=512, norm=1.000)"
2,512900,Streets of Rogue,"ndarray(shape=512, norm=1.000)"
1,646910,The Crew 2,"ndarray(shape=512, norm=1.000)"
0,753420,Dungreed,"ndarray(shape=512, norm=1.000)"


Additional samples:
  app_id=753420: ndarray(shape=512, norm=1.000)
  app_id=646910: ndarray(shape=512, norm=1.000)
  app_id=512900: ndarray(shape=512, norm=1.000)



### `keywords_names`

,dtype,coverage_pct,value_kind
0,object,91.11,array


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,"array(['roguelike', 'procedural generation', '..."
1,646910,The Crew 2,"array(['cars', 'planes', 'driving', 'sequel', ..."
0,753420,Dungreed,<empty>


Additional samples:
  app_id=646910: array(['cars', 'planes', 'driving', 'sequel', 'xbox one x enhanced'])
  app_id=512900: array(['roguelike', 'procedural generation', 'action-adventure', 'procedurally generated', 'pax west 2017', 'pax south 2017', 'pax west 2016', 'roguelite', 'speedrun mode'])
  app_id=637090: array(['post-apocalyptic', 'mech', 'turn-based', 'robots', 'war', 'apocalypse', 'campaign', 'mecha', 'crowdfunding - kickstarter', 'digital distribution', 'tactical turn-based combat', 'rivaling factions', 'customizable characters', 'pax west 2016', 'player vs player', 'mercenary', 'pax prime 2015', 'destructible environment', 'ancient advanced civilization technology', 'feudalism', 'negotiation', 'available on - pc gamepass'])



### `keywords_names__use`

,dtype,coverage_pct,value_kind
0,object,91.11,array


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,"array([array([ 0.00967012, 0.03305227, -0.023..."
1,646910,The Crew 2,"array([array([-5.69431260e-02, -1.09716905e-02..."
0,753420,Dungreed,<empty>


Additional samples:
  app_id=646910: array([array([-5.69431260e-02, -1.09716905e-02,  4.19170223e-02,  3.32025848e-02,
       -3.39245722e-02,  1.54240336e-02, -4.20597708e-03, -4.85024564e-02,
       -7.74668977e-02,  6.41510915e-03,  2.55251806e-02,  2.57690400e-02,
       -1.08237965e-02,  3.35025694e-03, -1.09149544e-02, -6.08008280e-02,
        2.17184741e-02,  8.26481078e-03, -9.97642707e-03, -2.08640918e-02,
        4.03512307e-02, -2.28972584e-02, -3.46010216e-02, -7.07664900e-03,
        5.02330959e-02, -4.25775871e-02, -1.12252319e-02, -7.06521645e-02,
        3.10222851e-03,  6.54099584e-02, -3.91539298e-02, -3.91735509e-02,
       -5.98454550e-02, -6.16864748e-02, -4.63748015e-02,  1.10378284e-02,
        8.62851832e-03, -4.33480293e-02,  5.38662449e-02,  5.45926206e-02,
        7.52650946e-02,  3.31793055e-02, -8.26150253e-02, -6.87812120e-02,
       -8.92773345e-02,  7.69763887e-02, -5.23068048e-02,  2.90335454e-02,
       -2.03943644e-02,  4.16965112e-02, -4.23804745e-02

### `keywords_names__use_pooled` **V2_CORE**

**Job 2 note:** Mean-pooled, L2-normalized USE vector over resolved taxonomy entity names.

,dtype,coverage_pct,value_kind,embed_dim
0,object,91.11,embedding,512


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,"ndarray(shape=512, norm=1.000)"
1,646910,The Crew 2,"ndarray(shape=512, norm=1.000)"
0,753420,Dungreed,<empty>


Additional samples:
  app_id=646910: ndarray(shape=512, norm=1.000)
  app_id=512900: ndarray(shape=512, norm=1.000)
  app_id=637090: ndarray(shape=512, norm=1.000)



### `game_modes_names`

,dtype,coverage_pct,value_kind
0,object,100.0,array


,app_id,app_name,sample
314,431960,Wallpaper Engine,array(['Single player'])
2,512900,Streets of Rogue,"array(['Single player', 'Multiplayer', 'Co-ope..."
1,646910,The Crew 2,"array(['Single player', 'Multiplayer', 'Co-ope..."
0,753420,Dungreed,array(['Single player'])


Additional samples:
  app_id=753420: array(['Single player'])
  app_id=646910: array(['Single player', 'Multiplayer', 'Co-operative', 'Massively Multiplayer Online (MMO)'])
  app_id=512900: array(['Single player', 'Multiplayer', 'Co-operative', 'Split screen'])



### `game_modes_names__use`

,dtype,coverage_pct,value_kind
0,object,100.0,array


,app_id,app_name,sample
314,431960,Wallpaper Engine,"array([array([-0.03297699, -0.05110957, -0.079..."
2,512900,Streets of Rogue,"array([array([-0.03297699, -0.05110957, -0.079..."
1,646910,The Crew 2,"array([array([-0.03297699, -0.05110957, -0.079..."
0,753420,Dungreed,"array([array([-0.03297699, -0.05110957, -0.079..."


Additional samples:
  app_id=753420: array([array([-0.03297699, -0.05110957, -0.07943583,  0.0157402 ,  0.03807794,
        0.07262456, -0.0045171 ,  0.00207252, -0.00558094,  0.01185589,
        0.02590894,  0.04430543,  0.00183716,  0.00956412,  0.03745565,
        0.05598426,  0.00403697,  0.03264809, -0.05205435, -0.00661937,
        0.00123201, -0.05215333,  0.02031393, -0.01282335,  0.03055757,
       -0.04274064,  0.05290025, -0.04281396,  0.06537234,  0.04149732,
        0.00270977, -0.05485355, -0.00571896,  0.07838769,  0.07029407,
        0.04555925, -0.01988393,  0.01232094, -0.01293933, -0.04790446,
       -0.03142112,  0.0461706 , -0.01367983, -0.06443056, -0.02689481,
        0.05405064, -0.05407386, -0.05566888, -0.02226171,  0.04039835,
        0.0913516 ,  0.02418532,  0.07846051,  0.0723381 ,  0.01305194,
        0.03088074,  0.0071243 ,  0.02497753,  0.06828979, -0.0269437 ,
        0.02267435,  0.0051435 , -0.0393327 ,  0.03282378, -0.05568551,
        0.00490273, 

### `game_modes_names__use_pooled` **V2_CORE**

**Job 2 note:** Mean-pooled, L2-normalized USE vector over resolved taxonomy entity names.

,dtype,coverage_pct,value_kind,embed_dim
0,object,100.0,embedding,512


,app_id,app_name,sample
314,431960,Wallpaper Engine,"ndarray(shape=512, norm=1.000)"
2,512900,Streets of Rogue,"ndarray(shape=512, norm=1.000)"
1,646910,The Crew 2,"ndarray(shape=512, norm=1.000)"
0,753420,Dungreed,"ndarray(shape=512, norm=1.000)"


Additional samples:
  app_id=753420: ndarray(shape=512, norm=1.000)
  app_id=646910: ndarray(shape=512, norm=1.000)
  app_id=512900: ndarray(shape=512, norm=1.000)



### `player_perspectives_names`

,dtype,coverage_pct,value_kind
0,object,95.87,array


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,array(['Bird view / Isometric'])
1,646910,The Crew 2,array(['Third person'])
0,753420,Dungreed,array(['Side view'])


Additional samples:
  app_id=753420: array(['Side view'])
  app_id=646910: array(['Third person'])
  app_id=512900: array(['Bird view / Isometric'])



### `player_perspectives_names__use`

,dtype,coverage_pct,value_kind
0,object,95.87,array


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,"array([array([ 1.15429936e-02, 3.00333817e-02..."
1,646910,The Crew 2,"array([array([-1.52518824e-02, -4.75310115e-03..."
0,753420,Dungreed,"array([array([-0.04921723, 0.05141259, -0.055..."


Additional samples:
  app_id=753420: array([array([-0.04921723,  0.05141259, -0.05595957,  0.06109421,  0.02894412,
        0.04448515, -0.01554719,  0.08353405,  0.01071263,  0.06216782,
       -0.03339449, -0.04917073, -0.04560273,  0.06908646, -0.00147597,
        0.05056075,  0.00928767,  0.02807641, -0.0481313 ,  0.00643074,
        0.04883599,  0.06424363,  0.01323038, -0.00839895,  0.01809941,
        0.01962275,  0.05653445,  0.01959766,  0.03115055,  0.01278651,
       -0.05058283,  0.01183613,  0.08814511,  0.08458356,  0.02501506,
       -0.00934457,  0.036969  ,  0.09585288,  0.02655973,  0.03174154,
        0.04761412,  0.04089251,  0.0127559 , -0.08415   , -0.04212987,
        0.03502839, -0.0525949 , -0.0233514 ,  0.05910507,  0.03148343,
        0.03616824,  0.07245543,  0.07684159,  0.05755792, -0.03728059,
       -0.0033486 , -0.0575437 ,  0.02474313,  0.08491006,  0.0718238 ,
        0.01872151, -0.00525063, -0.03834005,  0.07073191, -0.03743332,
       -0.02047874, 

### `player_perspectives_names__use_pooled` **V2_CORE**

**Job 2 note:** Mean-pooled, L2-normalized USE vector over resolved taxonomy entity names.

,dtype,coverage_pct,value_kind,embed_dim
0,object,95.87,embedding,512


,app_id,app_name,sample
314,431960,Wallpaper Engine,<empty>
2,512900,Streets of Rogue,"ndarray(shape=512, norm=1.000)"
1,646910,The Crew 2,"ndarray(shape=512, norm=1.000)"
0,753420,Dungreed,"ndarray(shape=512, norm=1.000)"


Additional samples:
  app_id=753420: ndarray(shape=512, norm=1.000)
  app_id=646910: ndarray(shape=512, norm=1.000)
  app_id=512900: ndarray(shape=512, norm=1.000)



# Key Findings

**Job 1 — `lookups/games.parquet` (315 rows)**

| V2 field | Coverage | Shape | V2 role |
|----------|----------|-------|---------|
| `genres` | 100% | id_list (median 3 ids) | V2a Jaccard |
| `themes` | 98.4% | id_list | V2a Jaccard |
| `keywords` | 91.1% | id_list | V2a Jaccard |
| `game_modes` | 100% | id_list | V2a Jaccard |
| `player_perspectives` | 95.9% | id_list | V2a Jaccard |
| `summary` / `summary__use` | 100% | text / 512-d vec | V2b text sim |
| `storyline` / `storyline__use` | 100% | text / 512-d vec | V2b (secondary) |
| `franchises` | 29.8% | id_list | Defer |

**Job 2 — `igdb_games__enriched.parquet`**

- All five taxonomy fields have `{field}_names`, `{field}_names__use` (per-entity vector lists), and `{field}_names__use_pooled` (single 512-d mean-pooled vector).
- Pooled-column coverage matches FK coverage per field (e.g. `keywords_names__use_pooled` 91.1%).
- Wallpaper Engine mock (`431960`, `manual_mock`) participates in enriched output with resolved `genres_names` and `themes_names`.

**Decision:** Validations passed — **good to go on V2a** using FK id Jaccard on the five taxonomy fields above.